flow/project.py

```diff
@@ -1971,27 +1985,35 @@ def _print_progress(x):
                                               ignore_errors=ignore_errors,
                                               cached_status=cached_status)
 
-        groups = []
+        singleton_groups = []
         for group in self._groups.values():
             if len(group.operations) == 1:
                 for name in group.operations:
                     if name == group.name:
-                        groups.append(group)
-
-        op_results = list(tqdm(
-            iterable=map(_get_group_status, groups),
-            desc="Collecting job status per operation.",
-            total=len(groups)))
+                        singleton_groups.append(group)
+
+        def _generate_results_with_tqdm(iterable, map=None, desc=None, len_itr=None):
```

Comment:
I think having `iterable`, `func`, and `map` be arguments would be better. Then we can just pass `singleton_gropups` to the function with `_get_group_status`.

Reply:
@b-butler we'd then have to check the name of the function too. Because if we don't do that then we have to pass description also.
And then this method won't be of much use as the code will be of the same length as before.

Reply:
Why would we need to check the name of the function too? We could just pass the function. I also, don't think passing the description is necessarily a bad thing. We could use partial functions to fill out what is known beforehand and just call the partial at sight with the necessary arguments.

Reply:
I didn't yet resolve this, but I will do this most probably by today